### ISSUE TITLE
Download command outputs partial threading output with log level error

### ISSUE THREAD

**Issue by jgeorgeson** [JFrogDev/jfrog-cli-go#30]
I expected a successful download to have no output when setting env variable `JFROG_CLI_LOG_LEVEL=ERROR`. However some, but not all, of the threading output is still printed.

### Normal log level stripping threading output with grep

```shell
$ jfrog rt dl p2-lgc/dsg/thirdparty/master/10.5.4-Drop00/ --flat=false | grep -v Thread
[Info:] Searching Artifactory using AQL query: items.find({"repo": "p2-lgc","$or": [{"$and": [{"path": {"$match":"dsg/thirdparty/master/10.5.4-Drop00"},"name":{"$match":"*"}}]},{"$and": [{"path": {"$match":"dsg/thirdparty/master/10.5.4-Drop00/*"},"name":{"$match":"*"}}]}]}).include("name","repo","path","actual_md5","actual_sha1","size")
[Info:] Artifactory response: 200 OK
[Info:] Found 345 artifacts.
[Info:] Downloaded 345 artifacts from Artifactory.
```

### ERROR output with no grep filtering

```shell
$ JFROG_CLI_LOG_LEVEL=ERROR jfrog rt dl p2-lgc/dsg/thirdparty/master/10.5.4-Drop00/ --flat=false
[Thread 2] [2]: 206 Partial Content...                                                          
[Thread 2] [1]: 206 Partial Content...                                                          
[Thread 2] [0]: 206 Partial Content...                                                          
[Thread 0] [1]: 206 Partial Content...                                                          
[Thread 0] [2]: 206 Partial Content...                                                          
[Thread 0] [0]: 206 Partial Content...                                                          
[Thread 1] [2]: 206 Partial Content...                                                          
[Thread 1] [1]: 206 Partial Content...                                                          
[Thread 1] [0]: 206 Partial Content...                                                          
[Thread 1] Done downloading.                                                                    
[Thread 1] [2]: 206 Partial Content...                                                          
[Thread 1] [1]: 206 Partial Content...                                                          
[Thread 1] [0]: 206 Partial Content...                                                          
[Thread 1] Done downloading.                                                                    
[Thread 1] [1]: 206 Partial Content...                                                          
[Thread 1] [0]: 206 Partial Content...                                                          
[Thread 1] [2]: 206 Partial Content...                                                          
[Thread 1] Done downloading.                                                                    
[Thread 1] [0]: 206 Partial Content...                                                          
[Thread 1] [2]: 206 Partial Content...                                                          
[Thread 1] [1]: 206 Partial Content...                                                          
[Thread 1] Done downloading.                                                                    
[Thread 0] Done downloading.                                                                    
[Thread 2] Done downloading.                                                                    
[Thread 1] [0]: 206 Partial Content...                                                          
[Thread 1] [2]: 206 Partial Content...                                                          
[Thread 1] [1]: 206 Partial Content...                                                          
[Thread 1] Done downloading.                                                                    
[Thread 2] [2]: 206 Partial Content...                                                          
[Thread 2] [0]: 206 Partial Content...                                                          
[Thread 2] [1]: 206 Partial Content...                                                          
[Thread 1] [0]: 206 Partial Content...                                                          
[Thread 1] [2]: 206 Partial Content...                                                          
[Thread 1] [1]: 206 Partial Content...                                                          
[Thread 1] Done downloading.                                                                    
[Thread 1] [0]: 206 Partial Content...                                                          
[Thread 1] [2]: 206 Partial Content...                                                          
[Thread 1] [1]: 206 Partial Content...                                                          
[Thread 0] [0]: 206 Partial Content...                                                          
[Thread 0] [2]: 206 Partial Content...                                                          
[Thread 0] [1]: 206 Partial Content...                                                          
[Thread 2] Done downloading.                                                                    
[Thread 0] Done downloading.                                                                    
[Thread 0] [0]: 206 Partial Content...                                                          
[Thread 0] [2]: 206 Partial Content...                                                          
[Thread 0] [1]: 206 Partial Content...                                                          
[Thread 0] Done downloading.                                                                    
[Thread 0] [1]: 206 Partial Content...                                                          
[Thread 0] [0]: 206 Partial Content...                                                          
[Thread 0] [2]: 206 Partial Content...                                                          
[Thread 1] Done downloading.                                                                    
[Thread 0] Done downloading.                                                                    
[Thread 2] [2]: 206 Partial Content...
[Thread 2] [1]: 206 Partial Content...
[Thread 2] [0]: 206 Partial Content...
[Thread 0] [1]: 206 Partial Content...
[Thread 0] [2]: 206 Partial Content...
[Thread 0] [0]: 206 Partial Content...
[Thread 1] [2]: 206 Partial Content...
[Thread 1] [1]: 206 Partial Content...
[Thread 1] [0]: 206 Partial Content...
[Thread 2] Done downloading.
[Thread 2] [0]: 206 Partial Content...
[Thread 2] [1]: 206 Partial Content...
[Thread 2] [2]: 206 Partial Content...
[Thread 1] Done downloading.
[Thread 2] Done downloading.
[Thread 1] [1]: 206 Partial Content...
[Thread 1] [2]: 206 Partial Content...
[Thread 1] [0]: 206 Partial Content...
[Thread 0] Done downloading.
[Thread 0] [0]: 206 Partial Content...
[Thread 0] [1]: 206 Partial Content...
[Thread 0] [2]: 206 Partial Content...
[Thread 1] Done downloading.
[Thread 0] Done downloading.
[Thread 2] [2]: 206 Partial Content...
[Thread 2] [0]: 206 Partial Content...
[Thread 2] [1]: 206 Partial Content...
[Thread 1] [2]: 206 Partial Content...
[Thread 1] [0]: 206 Partial Content...
[Thread 1] [1]: 206 Partial Content...
[Thread 2] Done downloading.
[Thread 1] Done downloading.
[Thread 0] [2]: 206 Partial Content...
[Thread 0] [1]: 206 Partial Content...
[Thread 0] [0]: 206 Partial Content...
[Thread 1] [0]: 206 Partial Content...
[Thread 1] [1]: 206 Partial Content...
[Thread 1] [2]: 206 Partial Content...
[Thread 1] Done downloading.
[Thread 0] Done downloading.
```

**Comment by eyalbe4** [comment]
@jgeorgeson, since *JFROG_CLI_LOG_LEVEL* is an environment variable, please use:
*export JFROG_CLI_LOG_LEVEL=ERROR*
or 
*set JFROG_CLI_LOG_LEVEL=ERROR*
depending on your operating system.

**Comment by jgeorgeson** [comment]
Bash allows setting env variables inline on the command.

```shell
$ export JFROG_CLI_LOG_LEVEL=ERROR
$ jfrog rt dl p2-lgc/dsg/thirdparty/master/10.5.4-Drop00/ --flat=false
[Thread 2] [1]: 206 Partial Content...
[Thread 2] [2]: 206 Partial Content...
[Thread 2] [0]: 206 Partial Content...
[Thread 0] [1]: 206 Partial Content...
[Thread 0] [2]: 206 Partial Content...
[Thread 0] [0]: 206 Partial Content...
[Thread 1] [0]: 206 Partial Content...
......
```

**Comment by eyalbe4** [comment]
@jgeorgeson,
We've just tried the following with version 1.5.1 of the CLI on both CentOS and Mac:
*JFROG_CLI_LOG_LEVEL=ERROR jfrog rt dl libs-release-local --flat=false*
It seems to be working.
Is there a chance you're using an old version of JFrog CLI?

**Comment by jgeorgeson** [comment]
Nope, I am on RHEL 6.8 though

```
$ jfrog -v
jfrog version 1.5.1
$ lsb_release -a
LSB Version:    :base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch:graphics-4.0-amd64:graphics-4.0-noarch:printing-4.0-amd64:printing-4.0-noarch
Distributor ID: RedHatEnterpriseWorkstation
Description:    Red Hat Enterprise Linux Workstation release 6.8 (Santiago)
Release:        6.8
Codename:       Santiago
```

**Comment by eyalbe4** [comment]
@jgeorgeson,, the CLI code retrieves the env var value as follows:
*os.Getenv("JFROG_CLI_LOG_LEVEL")*
I'm not sure how this API behaves on RHEL when the variable is set inline the shell command.

**Comment by jgeorgeson** [comment]
I see the same output regardless of manually exporting the env variable before running the jfrog command vs specifying the variable inline. I just tested on a CentOS 7 machine and it's working as expected there. If you can test on CentOS 6.x I suspect you'll see the same behavior that I'm reporting on RHEL 6.

**Comment by eyalbe4** [comment]
@jgeorgeson,
Although I couldn't find any documentation or article discussing this issue, following your testing, it looks like golang's os.Getenv API can't retrieve the environment variable, when it is defined inline, specifically on RHEL. It looks like there's nothing we can do in the CLI code to resolve this. Please let us know what you think.

**Comment by eyalbe4** [comment]
@jgeorgeson, with your permission, I'm closing this issue.

**Comment by jgeorgeson** [comment]
I wasn't claiming the `JFROG_CLI_LOG_LEVEL` variable is having no effect. It's having a _partial_ effect on RedHat/CentOS 6.x. 

```
[hbl4218@woodhouse: /d01/sandboxes/hbl4218/workspaces/jfrog-cli]
$ export JFROG_CLI_LOG_LEVEL=ERROR
[hbl4218@woodhouse: /d01/sandboxes/hbl4218/workspaces/jfrog-cli]
$ jfrog rt dl p2-lgc/dsg/thirdparty/master/10.5.4-Drop00/       
[Thread 0] [0]: 206 Partial Content...                          
[Thread 0] [1]: 206 Partial Content...                          
[Thread 1] [2]: 206 Partial Content...                          
[Thread 1] [0]: 206 Partial Content...                          
[Thread 1] [1]: 206 Partial Content...                          
[Thread 0] [2]: 206 Partial Content...                          
[Thread 2] [1]: 206 Partial Content...                          
[Thread 2] [0]: 206 Partial Content...                          
[Thread 2] [2]: 206 Partial Content...                          
[Thread 2] Done downloading.                                    
[Thread 2] [1]: 206 Partial Content...         
```

vs

```
[hbl4218@woodhouse: /d01/sandboxes/hbl4218/workspaces/jfrog-cli]
$ unset JFROG_CLI_LOG_LEVEL
[hbl4218@woodhouse: /d01/sandboxes/hbl4218/workspaces/jfrog-cli]
$ jfrog rt dl p2-lgc/dsg/thirdparty/master/10.5.4-Drop00/
[Info] Searching artifacts...                            
[Info] [Thread 1] Downloading p2-lgc/dsg/thirdparty/master/10.5.4-Drop00/binary/com.lgc.gis.thirdparties.feature_root.win32.win32.x86_64_10.5.4.20161102221513-ff81f86018                                                                                                         
[Info] [Thread 2] Downloading p2-lgc/dsg/thirdparty/master/10.5.4-Drop00/binary/com.lgc.gis.thirdparties.feature_root.gtk.linux.x86_64_10.5.4.20161102221513-ff81f86018                                                                                                           
[Info] [Thread 0] Downloading p2-lgc/dsg/thirdparty/master/10.5.4-Drop00/artifacts.jar                                                   
[Info] [Thread 0] Downloading p2-lgc/dsg/thirdparty/master/10.5.4-Drop00/binary/com.lgc.gis.thirdparties.feature_root_10.5.4.20161102221513-ff81f86018                                                                                                                            
[Thread 1] [0]: 206 Partial Content...                                                                                                   
[Thread 1] [1]: 206 Partial Content...                                                                                                   
[Thread 2] [1]: 206 Partial Content...                                                                                                   
[Thread 1] [2]: 206 Partial Content...                                                                                                   
[Thread 2] [0]: 206 Partial Content...                                                                                                   
[Info] [Thread 0] Downloading p2-lgc/dsg/thirdparty/master/10.5.4-Drop00/content.jar                                                     
[Thread 2] [2]: 206 Partial Content...                                                                                                   
[Info] [Thread 0] Downloading p2-lgc/dsg/thirdparty/master/10.5.4-Drop00/features/au.com.bytecode.opencsv.feature_2.3.0.43890.jar        
[Info] [Thread 0] Downloading p2-lgc/dsg/thirdparty/master/10.5.4-Drop00/features/bsh.feature_2.0.4.67231.jar                            
```

So we can see the full output without the `JFROG_CLI_LOG_LEVEL` unset, and partial output with `JFROG_CLI_LOG_LEVEL=ERROR`. Running on  RedHat/CentOS 7 I see _no_ output with `JFROG_CLI_LOG_LEVEL=ERROR`.

RedHat, and by extension CentOS, 6.x still have years of official support remaining. Could you please add that platform to your testing environment. There's plenty of Docker and Vagrant images publicly available that should making testing things like the CLI and [JFrogDev/artifactory-scripts](/JFrogDev/artifactory-scripts) easy and automatable.

**Comment by eyalbe4** [comment]
Thanks for the clarification @jgeorgeson.
We pushed [this](https://github.com/JFrogDev/jfrog-cli-go/commit/d285e328c018f5f1533270a3952a3539c275bce7) commit to the *dev* branch, which fixes this (*fmt.Println* was replaced with *log.Info*). This will be included as part of the next release.

**Comment by eyalbe4** [comment]
@jgeorgeson, this fix made it to version 1.8.0 of JFrog CLI.
I'm therefore closing this issue.
